torch/testing/_internal/common_utils.py

```diff
@@ -2495,16 +2495,17 @@ def random_fullrank_matrix_distinct_singular_value(matrix_size, *batch_dims,
 
 # Creates a full rank matrix with distinct signular values or
 #   a batch of such matrices
-# Shape must be a square matrix or batch of square matrices
-def make_fullrank_matrices_with_distinct_singular_values(*shape, device, dtype):
-    assert shape[-1] == shape[-2]
-    t = make_tensor(shape, device=device, dtype=dtype)
-    u, _, vh = torch.linalg.svd(t, full_matrices=False)
-    # TODO: improve the handling of complex tensors here
-    real_dtype = t.real.dtype if t.dtype.is_complex else t.dtype
-    s = torch.arange(1., shape[-1] + 1, dtype=real_dtype, device=device).mul_(1.0 / (shape[-1] + 1))
-    return (u * s.to(dtype)) @ vh
-
+def make_fullrank_matrices_with_distinct_singular_values(*shape, device, dtype, requires_grad=False):
+    with torch.no_grad():
```

Comment:
Nice refinement

Reply:
fwiw, there's a PR later in this stack that improves even further this function. After discussing it with the usual suspects, they were happy with me shortening its name and changing its signature to take a tuple.

Reply:
Making functions like this have signatures more consistent with make_tensor is great. Eventually it might be interesting to expose them in torch.testing. We could make them accept just tuples or tuples and a splatted shape if you like.

Reply:
I like the idea of them accepting just tuples. This would make the code more consistent all across.